### DISCUSSION TITLE
Feature Request: Ability to Cancel Query from Python

### DISCUSSION THREAD

**Discussion by totalhack** [duckdb/duckdb#6850] [Ideas]
I'm trying to integrate duckdb in another analytics library that needs the ability to cancel queries if the user aborts the report. Given a reference to the connection that started the query I'd like to be able to interrupt/cancel that query.

I've seen [this test](https://github.com/Tishj/duckdb/blob/bc9afc6e00476873dba76c094596bf27783ed33f/tools/pythonpkg/tests/fast/api/test_query_interrupt.py) which shows that a keyboard interrupt should do it, but a similar approach of getting KeyboardInterrupt raised in the thread that started the query didn't work (using `async_raise` from `stopit`).

Other connectors such as psycopg2 or sqlite have methods to cancel/interrupt.

**Comment by Tishj** [comment]
issue #5938
